qiskit/providers/fake_provider/fake_generic.py

```diff
@@ -0,0 +1,398 @@
+                    distance=distance, bidirectional=bidirectional_cp_mp
+                )
+
+            elif self._coupling_map_type == "grid":
+                num_rows, num_columns = self._get_cmap_args(num_qubits=num_qubits)
+                coupling_map = CouplingMap().from_grid(
+                    num_rows=num_rows, num_columns=num_columns, bidirectional=bidirectional_cp_mp
+                )
+        else:
+            coupling_map = CouplingMap(coupling_map)
+
+        num_qubits = coupling_map.size()
+
+        self._target = Target(
+            description="Fake Generic Backend",
+            num_qubits=num_qubits,
+            dt=dt,
+            qubit_properties=[
+                QubitProperties(
+                    t1=self._rng.uniform(100e-6, 200e-6),
+                    t2=self._rng.uniform(100e-6, 200e-6),
+                    frequency=self._rng.uniform(5e9, 5.5e9),
+                )
+                for _ in range(num_qubits)
+            ],
+        )
+
+        instruction_dict = self._get_instruction_dict(num_qubits, coupling_map)
+        for gate in self.basis_gates:
+            try:
+                self._target.add_instruction(*instruction_dict[gate])
+            except:
+                raise QiskitError(f"{gate} is not a valid basis gate")
+
+        if dynamic:
+            self._target.add_instruction(IfElseOp, name="if_else")
+            self._target.add_instruction(WhileLoopOp, name="while_loop")
+            self._target.add_instruction(ForLoopOp, name="for_loop")
+            self._target.add_instruction(SwitchCaseOp, name="switch_case")
+            self._target.add_instruction(BreakLoopOp, name="break")
+            self._target.add_instruction(ContinueLoopOp, name="continue")
+
+        if enable_reset:
+            self._target.add_instruction(
+                Reset(), {(qubit_idx,): None for qubit_idx in range(num_qubits)}
+            )
+
+        defaults = self._build_calibration_defaults(skip_calibration_gates)
+
+        inst_map = defaults.instruction_schedule_map
+        for inst in inst_map.instructions:
+            for qarg in inst_map.qubits_with_instruction(inst):
+                try:
+                    qargs = tuple(qarg)
+                except TypeError:
+                    qargs = (qarg,)
+                # Do NOT call .get method. This parses Qpbj immediately.
+                # This operation is computationally expensive and should be bypassed.
+                calibration_entry = inst_map._get_calibration_entry(inst, qargs)
+                if inst in self._target._gate_map:
+                    if inst == "measure":
+                        for qubit in qargs:
+                            self._target._gate_map[inst][(qubit,)].calibration = calibration_entry
+                    elif qargs in self._target._gate_map[inst] and inst != "delay":
+                        self._target._gate_map[inst][qargs].calibration = calibration_entry
```

Comment:
The only solution I have found so far to add calibration schedules in the format that is expected by the unit tests is pretty convoluted, because the only reference I have are configurations that are imported from json files. What I do here is: 

1. I build calibration defaults using the helped method from `fake_provider/utils/configurable_backend.py` (see line 310)
2. I convert them into calibration entries as if we were importing these defaults from a json file

I am sure there must be a more direct way of directly creating these "calibration entries" without them being detected as "provided by user", but I wanted to try first to see if I could get the tests to pass at all, which they seem to do. Conclusion: to be improved.